WebSocket beforeHandle unable to access use'd plugins / state / derive's

I'd like to have the ability to check for certain cookies before allowing a WebSocket connection, as well as have access to the Elysia context. Here's some demo code:

```ts
import { Elysia, ws } from 'elysia'
import cookie from '@elysiajs/cookie';

new Elysia()
	.use(cookie())
	.use(ws())
	.state('count', 0)
	.ws('/ws', {
		beforeHandle({ cookie, store }) {
			if (!cookie['authToken']) return new Response(null, { status: 401 });

			console.log(store.count);
		}
	})
	.listen(8080)
```

When I run `tsc`, I get these errors for `cookie` in the params and `store.count`:
```
index.ts:9:18 - error TS2339: Property 'cookie' does not exist on type '{ request: Request; headers: Record<string, string | null>; query: Record<string, unknown>; params: Record<never, string>; body: unknown; store: {}; set: { headers: Record<...>; status?: number | undefined; redirect?: string | undefined; }; } & Record<...>'.

9   beforeHandle({ cookie, store }) {
                   ~~~~~~

index.ts:12:22 - error TS2339: Property 'count' does not exist on type '{}'.

12    console.log(store.count);
                        ~~~~~


Found 2 errors in the same file, starting at: index.ts:9
```

Ideally, I should be able to access anything available on the context in these WebSocket handlers as I would in regular HTTP endpoints.

I'm currently using Bun `0.6.8`, and `bun pm ls` outputs
```
├── @elysiajs/cookie@0.5.2
├── bun-types@0.6.9
├── elysia@0.5.19
└── typescript@5.1.3
```

EDIT: This might not be relevant for right now, but I'd like to see the notion of a derive in the context of WebSockets, e.g. some way to add data to the "ws" objects passed into open / message / close. This'd be real handy if I wanted to attach certain data, such as user IDs.

--- a/src/index.ts
+++ b/src/index.ts
@@ -2797,7 +2797,7 @@ export default class Elysia<
 			? ElysiaWSOptions<
 					`${BasePath}${Path}`,
 					Schema,
-					Instance['meta']['defs']
+					Instance
 			  >
 			: never
 	): Elysia<

--- a/src/ws/types.ts
+++ b/src/ws/types.ts
@@ -70,23 +70,23 @@ export type TransformMessageHandler<Message extends WSTypedSchema['body']> = (
 
 export type ElysiaWSContext<
 	Schema extends WSTypedSchema<any> = WSTypedSchema<never>,
-	Definitions extends ElysiaInstance['meta']['defs'] = {},
+	Instance extends ElysiaInstance = ElysiaInstance,
 	Path extends string = never
 > = ServerWebSocket<
 	Context<{
-		body: UnwrapSchema<Schema['body'], Definitions>
-		headers: UnwrapSchema<Schema['headers'], Definitions, Record<string, string>>
-		query: UnwrapSchema<Schema['query'], Definitions, Record<string, string>>
+		body: UnwrapSchema<Schema['body'], Instance['meta']['defs']>
+		headers: UnwrapSchema<Schema['headers'], Instance['meta']['defs'], Record<string, string>>
+		query: UnwrapSchema<Schema['query'], Instance['meta']['defs'], Record<string, string>>
 		params: ExtractPath<Path> extends infer Params extends string
 			? Record<Params, string>
-			: UnwrapSchema<Schema['params'], Definitions, Record<string, string>>
-		response: UnwrapSchema<Schema['response'], Definitions>
+			: UnwrapSchema<Schema['params'], Instance['meta']['defs'], Record<string, string>>
+		response: UnwrapSchema<Schema['response'], Instance['meta']['defs']>
 	}> & {
 		id: number
 		message: TypeCheck<any>
 		transformMessage: TransformMessageHandler<Schema['body']>[]
 		schema: TypedRoute
-	}
+	} & Instance["request"]
 >
 
 export type WebSocketHeaderHandler<
@@ -110,12 +110,12 @@ export type WebSocketHeaderHandler<
 export type ElysiaWSOptions<
 	Path extends string,
 	Schema extends WSTypedSchema<any>,
-	Definitions extends ElysiaInstance['meta']['defs']
+	Instance extends ElysiaInstance
 > = Omit<
 	Partial<WebSocketHandler<Context>>,
 	'open' | 'message' | 'close' | 'drain' | 'publish' | 'publishToSelf'
 > &
-	(ElysiaWS<ElysiaWSContext<Schema, Definitions, Path>> extends infer WS
+	(ElysiaWS<ElysiaWSContext<Schema, Instance, Path>> extends infer WS
 		? Partial<Schema> & {
 				beforeHandle?: WithArray<HookHandler<Schema>>
 				transform?: WithArray<
@@ -147,7 +147,7 @@ export type ElysiaWSOptions<
 				 */
 				message?: (
 					ws: WS,
-					message: UnwrapSchema<Schema['body'], Definitions>
+					message: UnwrapSchema<Schema['body'], Instance['meta']['defs']>
 				) => any
 
 				/**

